raster-foundry/raster-foundry
# Update platform email settings

## Overview

This PR updates platform email settings so that:
* `emailFrom` is required and is validated on frontend to be in email format
* `setFrom()` uses `emailFrom` to make apache commons mail happy
* `emailSmtpUserName` is only used for authentication
* `emailSupport` is added to provide support email address instead of `noreply`

Paired with @rbreslow on this.

### Checklist


### Demo

![screen shot 2018-12-04 at 4 58 43 pm](https://user-images.githubusercontent.com/16109558/49477296-7c9ef400-f7ea-11e8-9951-a32f5535b89c.png)


### Notes

I think we should notify all platform admins after we deploy `1.16` so that they could go into the UI to update the email from field and support email field.

## Testing Instructions

 * Run migration
 * Log in as a member and set this account's notification preference
 * Log in as a platform admin
 * To test if this will work using AWS SES, grab the below platform email settings info from staging platform `Raster Foundry`:
    - `public_settings`
         - `emailFrom` 
         - `emailSmtpHost`
         - `emailSmtpEncryption`
         - `emailUser`, this should be used in the new `emailSmtpUserName` field, which is the `SMTP User Name` on the UI
     - `private_settings`
         - `emailPassword`
 * Go to UI `/admin/platform/<platformId>/settings/email` and populate fields with the above info
 * Create a new organization
 * Invite the member user from step two to join this organization
 * Check if the member user receives an email invitation
 * Test other actions like scene ingest, export, update aoi to trigger email notifications

Closes https://github.com/raster-foundry/raster-foundry/issues/4323
Closes https://github.com/azavea/raster-foundry-platform/issues/477
